drivers/pwm/pwm_nxp_flexio.c

```diff
@@ -130,18 +132,35 @@ static int pwm_nxp_flexio_set_cycles(const struct device *dev,
 
 	pwm_info = &config->pulse_info->pwm_info[channel];
 
-	if ((flags & PWM_POLARITY_INVERTED) == 0) {
+	polarity = (flags & PWM_POLARITY_INVERTED) == 0 ?
+		FLEXIO_PWM_ACTIVE_HIGH : FLEXIO_PWM_ACTIVE_LOW;
+
+	/*
+	 * Checking to see if duty cycle is 0% or 100%
+	 * If so manually keep the GPIO HIGH or LOW.
+	 */
+	if (period_cycles == pulse_cycles) {
+		polarity = FLEXIO_PWM_ACTIVE_LOW;
+		is_only_low_or_high = true;
+	} else if (duty_cycle == period_cycles && pulse_cycles == 0) {
 		polarity = FLEXIO_PWM_ACTIVE_HIGH;
+		is_only_low_or_high = true;
 	} else {
-		polarity = FLEXIO_PWM_ACTIVE_LOW;
+		/*
+		 * The duty cycle is in between 0% and 100%
+		 * so there is no need to force the line to be
+		 * low or high, the polarity can stay as
+		 * configured and the is_only_low_or_high
+		 * can remain false.
+		 */
```

Comment:
you don't need this else block if you just combine this `if` with the next one. I'm not sure why you are splitting this into two different `if` chains, it's confusing. All you are needing to do is this:

if (period_cycles == pulse_cycles) 
   ...
else if (pulse_cycles == 0)
   ...
else if (polarity == high)
  ...
else 
  ... <- polarity is low here then

Reply:
Updated